PriceCalculator.cs

```diff
@@ -1,25 +1,21 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Diagnostics;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-
-namespace Price_Calculator_kata
+﻿namespace Price_Calculator_kata
 {
     public class PriceCalculator
     {
         TaxServices MyTaxService;
-        public PriceCalculator(TaxServices MyTaxService) {
-           
+        DiscountService MyDiscountService;
+        public PriceCalculator(TaxServices MyTaxService, DiscountService MyDiscountService)
+        {
             this.MyTaxService = MyTaxService;
-        
-         }
-       
+            this.MyDiscountService = MyDiscountService;
+
+        }
+
         public double CalculateTotalPrice(Product product)
         {
             double taxAmount = CalculateTaxAmount(product);
-            double totalPrice = Math.Round(product.Price + taxAmount, 2);
+            double DiscountAmount = CalculateDiscountAmount(product);
+            double totalPrice = Math.Round(product.Price + taxAmount - DiscountAmount, 2);
```

Comment:
do you think this should be separated in a method? maybe for reusability ? here in the price calculator
WDUT?